### DISCUSSION TITLE
Why are variable values in the remote Jupyter server lost when reconnecting?

### DISCUSSION THREAD

**Discussion by Troublor** [microsoft/vscode-jupyter#9488] [Questions and Answers] [answered]
I have a python jupyter notebook server running on a server, and I use vscode to connect to the remote jupyter via the jupyter URI. 
However, the vscode cannot recognize any execution history of the current kernel session (I didn't restart the kernel). 

For instance, if I previously use the webpage UI of jupyter to execute some cells, the variables produced will be still there if I close the browser and reopen it. But this is not the case using vscode. In vscode, all previously computed variables are lost if I close vscode and I have had to re-execute again when I reconnect to the remote server, even if I didn't restart the kernel at all. Actually, those variables are still there if I use webpage UI, but they are not recognized in vscode. 

I need the variables to be preserved since I am working on a large project which I cannot get everything down in one day. 
I close the vscode at night and continue to work on it in the next morning. It is awful to re-execute everything from scratch every day. 

Is there any configuration of vscode that I am not aware of to make this work?

**Comment by rchiodo** [comment, marked as answer]
What's your version of VS code and the Jupyter extension? 

This should work automatically if you reopen the same notebook. We remember the remote kernel id and reconnect to it when opening the notebook.

However this was fixed/addressed rather recently. Can you try the latest prerelease bits with insiders to see if that works for you?

**Reply by Troublor** [reply]
Here's the version informations:
vscode: 
```
Version: 1.65.2
Commit: c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1
Date: 2022-03-10T14:33:49.188Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.4.0
```

vscode-jupyter:
```
pre-release
v2022.3.1000762011
```

OK. I see it works. The stable version started a lot of sessions with the same name, and when I first tried to use pre-release version, I was not aware of that and select the wrong one. 

Now I can confirm the pre-release version works. Thanks

**Reply by Troublor** [reply]
Could you point out the issue or commit that fix this issue?

**Reply by rchiodo** [reply]
This is the commit id:
https://github.com/microsoft/vscode-jupyter/commit/488474e9d8387159ee589bfa1508f2b9676e59d3
